EIPS/eip-6963.md

```diff
@@ -38,171 +38,189 @@ The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL
-
-The described orchestration of events guarantees that the DApp is able to discover the Wallet, regardless of which code executes first, the Wallet code or the DApp code.
+The Dapp MAY elect to persist the `Eip6963ProviderDetail` objects from `eip6963:announceProvider` events sent by multiple Wallet Providers. Thus, if the user wishes to utilize a different Wallet Provider within the same page session, the user can express this intent within the Dapp's interface, and the Dapp can immediately elect to send transactions to the newly-chosen Wallet Provider. Additionally, the Dapp MAY re-initiate the Wallet Provider discovery flow at any time by dispatching a new `eip6963:requestProvider` event, potentially discovering a different set of Wallet Providers.
 
 ## Rationale
 
-The previous proposal introduced mechanisms that relied on a single, mutable window object that could be overwritten by multiple parties. We opted for an event-based approach to avoid the race conditions, the namespace collisions, and the potential for "pollution" attacks on a shared mutable object; the event-based orchestration creates a bidirectional communication channel between wallet and dapp that can be re-orchestrated over time.
+The previous proposal ([EIP-1193][EIP-1193]) introduced mechanisms that relied on a single, mutable `window.ethereum` object that could be overwritten by multiple parties. To avoid race conditions, namespace collisions, and potential "pollution" attacks on a shared mutable object, we opted for an event-based approach that facilitates a bidirectional communication channel between Wallet Provider and Dapp that can be re-orchestrated multiple times within a browser session.
+
+The orchestration of events described above guarantees that the Dapp is able to discover the Wallet Provider, regardless of whether the Wallet Provider code or the Dapp code executes first. Additionally, multiple Wallet Providers can respond to requests from a single Dapp without polluting the `window.ethereum` namespace, which has historically resulted in non-deterministic behavior where a different Wallet Provider is connected for each page session.
 
-To follow the Javascript event name conventions, the names are written in present tense and are prefixed with the number of this document (`EIP6963`). 
+### Event names
+
+Event names are written in present tense in accordance with JavaScript conventions, and are prefixed with the number of this document (`eip6963:`) to avoid namespace collisions.
```

Comment:
i'm not sure i understand the insertion of the `:` namespacing symbol here?

Reply:
This is referring to the JavaScript event names `eip6963:requestProvider` and `eip6963:announceProvider`.